pkg/kube/controllers/boshdeployment/deployment_reconciler.go

```diff
@@ -209,9 +209,11 @@ func (r *ReconcileBOSHDeployment) createEJob(ctx context.Context, instance *bdv1
 
 	op, err := controllerutil.CreateOrUpdate(ctx, r.client, eJob.DeepCopy(), func(obj runtime.Object) error {
 		if existingEJob, ok := obj.(*ejv1.ExtendedJob); ok {
-			eJob.ObjectMeta.ResourceVersion = existingEJob.ObjectMeta.ResourceVersion
-			eJob.Spec.Trigger.Strategy = existingEJob.Spec.Trigger.Strategy
-			eJob.DeepCopyInto(existingEJob)
+			if shouldEJobUpdate(existingEJob, eJob) {
```

Comment:
`CreateOrUpdate` should be a no-op if the ejob exists and hasn't changed. We copy strategy and resource version to avoid unnecessary updates. So I guess there was a case where we did update the ejobs, even though we didn't have to?

Reply:
@manno Yes. After we copy strategy and resource version, CreateOrUpdate still update EJob.

Reply:
Seems like we have two strategies to avoid the unwanted update now. `shouldEJobUpdate` and the copying of the previous attributes.

Also I wasn't able to see CreateOrUpdate misbehave before in https://www.pivotaltracker.com/story/show/167037095

Reply:
We can see this action when we restart operator which mean that CreateOrUpdate did update for EJob resource:
```
2019-07-19T12:00:13.502+0800	DEBUG	boshdeployment-reconciler	boshdeployment/deployment_reconciler.go:212  	ExtendedJob 'bpm-nats-deployment' has been updated
```
IIRC, we didn't get operation results 10 days ago.